packages/browser-agent-core/common/config/state/runtime.js

```diff
@@ -3,23 +3,26 @@ import { getLastTimestamp } from '../../timing/now'
 import * as userAgent from '../../util/user-agent'
 import { Configurable } from './configurable'
 import { gosNREUMInitializedAgents } from '../../window/nreum'
+import { getCurrentSessionId } from '../../window/session-storage'
+import { getConfigurationValue } from '../config'
 
 var XHR = window.XMLHttpRequest
 var XHR_PROTO = XHR && XHR.prototype
 
-const model = {
-  origin: '' + window.location,
+const model = agentId => { return {
+  customTransaction: undefined,
+  disabled: false,
+  features: {},
   maxBytes: ieVersion === 6 ? 2000 : 30000,
   offset: getLastTimestamp(),
-  features: {},
-  customTransaction: undefined,
   onerror: undefined,
+  origin: '' + window.location,
+  ptid: undefined,
   releaseIds: {},
+  sessionId: getConfigurationValue(agentId, 'privacy.cookies_enabled') === true ? getCurrentSessionId() : '0',
```

Comment:
Just to play devil's advocate, can you think of any cases where we would need to get this sessionID after load time?  Like if the end-user clears their cache in the middle of a page session, are they expecting future payloads to not have that value? Do we need to generate a new one?

This is likely a non issue, but would like to confirm. cc: @aubreymasten

Reply:
Can you specify what cache you mean? Like all the state files' internal `_cache` per agent? Or the sessionStorage?
Wrt sessionStorage:
- It's true eu's can manually clear their storage, but if we're making an analogous reference to cookies, they could've wiped cookies mid-session too.
- In the worst case where `getCurrentSessionId()` is called again, and the sessionStorage has been wiped, it would just generate a new value for all future payloads (and store that again), so the session got effectively "lost" (interrupted). I think that makes sense?
- More interestingly, I want to discuss current behavior. Making the `sessionId` part of the per-agent runtime properties rather than retrieving it from SS each use does alleviate that concern a bit. In that case, the `sessionId` would retain the old value even after SS has been wiped, _until the time when `setRuntime()` is called again_. This relates to the way the default `model` and `Configurable()` was done, and I should mention at that time, it'll also be dependent on the then-value of privacy.cookies_enabled.

Reply:
```More interestingly, I want to discuss current behavior. Making the sessionId part of the per-agent runtime properties rather than retrieving it from SS each use does alleviate that concern a bit. In that case, the sessionId would retain the old value even after SS has been wiped, until the time when setRuntime() is called again. This relates to the way the default model and Configurable() was done, and I should mention at that time, it'll also be dependent on the then-value of privacy.cookies_enabled.```

Yep this is the part I was pointing to.  The behavior here would mean that if a client cleared their storage mid-session, we would still be sending "old" session data, which is probably fine.  Just wanted to confirm that that behavior is acceptable.

Reply:
Oh gotcha. I think that behavior makes sense, since the conventional `session` is defined per lifetime of a tab or window. If anything, what's wrong in that scenario is that _something_ (not just the e.u. but I think fellow scripts under the same domain can also `.clear()` the storage) is going in and destroying that definition.